honeybee_energy/measure.py

```diff
@@ -0,0 +1,428 @@
+
+        # parse any choice arguments if they exist
+        self._valid_choices = None
+        if self._type_text == 'Choice':
+            self._valid_choices = tuple(choice.find('value').text
+                                        for choice in xml_element.find('choices'))
+
+    @property
+    def identifier(self):
+        """Get text for the identifier of the argument.
+
+        This is also called the "name" in the measure.xml file.
+        """
+        return self._identifier
+
+    @property
+    def display_name(self):
+        """Get text for the human-readable display name of the argument.
+
+        This is called the "display_name" in the measure.xml file.
+        """
+        return self._display_name
+
+    @property
+    def value(self):
+        """Get or set the value for the argument.
+
+        If not set, this will be equal to the default_value and, if no default
+        value is included for this argument, it will be None.
+        """
+        if self._value:
+            return self._value
+        return self._default_value
+
+    @value.setter
+    def value(self, val):
+        assert isinstance(val, self._type), 'Value for measure argument "{}" ' \
+            'must be a {}. Got {}'.format(self.identifier, self._type, type(val))
+        if self._valid_choices:
+            assert val in self._valid_choices, 'Choice measure argument "{}" ' \
+                'must be one of the following:\n{}\nGot {}'.format(
+                    self.identifier, self._valid_choices, val)
+        self._value = val
+
+    @property
+    def default_value(self):
+        """Get the default value for the argument.
+
+        This may be None if no default value has been included.
+        """
+        return self._default_value
+
+    @property
+    def type(self):
+        """Get the Python type of argument this is (eg. float, str, int)."""
+        return self._type
+
+    @property
+    def type_text(self):
+        """Get a text string for the argument type as it appears in the measure.xml.
+
+        (eg. 'Double', 'String', 'Boolean')
+        """
+        return self._type_text
+
+    @property
+    def required(self):
+        """Get a boolean for whether this argument is required to run the measure."""
+        return self._required
+
+    @property
+    def description(self):
+        """Get text for describing what the measure does if it exists."""
+        return self._description
+
+    @property
+    def model_dependent(self):
+        """Get a boolean for whether this argument is depdent on the model."""
+        return self._model_dependent
+
+    @property
+    def valid_choices(self):
+        """Get a list of text for valid inputs for choice arguments.
+
+        This will be None if the argument type is not Choice.
+        """
+        return self._valid_choices
+
+    def validate(self, raise_exception=True):
+        """Check if the argument will cause errors when run through OpenStudio CLI.
```

Comment:
I would make this explanation more specific. In this case it looks like the check is to ensure all required arguments have a value. This is clearer than causing errors when run through OpenStudio CLI.